test/promise-cache.test.ts

```diff
@@ -45,6 +45,21 @@ describe("Promise Cache", () => {
         expectStats(cache, {hits: 1, misses: 1, entries: 1, failLoads: 0});
     });
 
+    it("can set cache", async () => {
+        const loader = new TestLoader("value", 5);
+        const cache = new PromiseCache<string>(() => loader.load());
+
+        cache.set("key", "value");
+
```

Comment:
I would rather check if the cache returns the correct result. Something like:
`const value = await cache.get("key");
 expect(value).to.eq("value");`

I think the rest of this method is not needed.

Reply:
I checked the stats instead of the value because the TestLoader would return "value" into the cache even if the set did not work. I could change the TestLoader to return a different value if you would prefer?

Reply:
Ah yes you are right. Or you could "set" another value. As you like. 

We could keep `expect(loader.timesLoaded).to.eq(0);` as well.

Reply:
I have changed the test to set the values as a Promise with a value which differs from the TestLoader value. I am not sure why there was not a compile error when setting the value as a string instead of a Promise.